Detect bindable resource based on status attribute

baijum opened the pull request:
Signed-off-by: Baiju Muthukadan <baiju.m.mail@gmail.com>

<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

<!-- 
Describe your changes here- ideally you can get that description straight from your descriptive commit message(s)! 

In addition, categorize the changes you're making using the "/kind" Prow command, example:

/kind <kind>

Supported kinds are: bug, test, documentation, enhancement
-->

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#docs) 
  included if any changes are user facing
- [ ] [Tests](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#tests)
  included if any functionality added or changed. For bugfixes please include tests that can catch regressions
- [ ] All acceptance test scenarios included in the PR which verifies a bugfix or a requested feature reported by a non-member are tagged with `@external-feedback` tag.
- [ ] Follows the [commit message standard](https://github.com/redhat-developer/service-binding-operator/blob/master/CONTRIBUTING.md#commits)

baijum commented:
Closing this in favour of #1259

sadlerap reviewed (commented):
LGTM so far, but I'd maybe like to see some tests (either some unit tests or some acceptance tests).  I'm a little suspicious of the JSONPath lookup; IMO it's something that could be fragile if we're not careful.

baijum commented:
> LGTM so far, but I'd maybe like to see some tests (either some unit tests or some acceptance tests). I'm a little suspicious of the JSONPath lookup; IMO it's something that could be fragile if we're not careful.

I have added unit tests. There is an existing acceptance test which I have updated the test data.

sadlerap reviewed (approved):
/lgtm
